docs/user/installconfig.schema.json

```diff
@@ -0,0 +1,417 @@
+  "title": "OpenShift install-config.yaml",
+  "description": "Installer configuration for an OpenShift cluster",
+  "type": "object",
+  "properties": {
+    "apiVersion": {
+      "description": "The API version for the install-config.yaml content. The current version is v1. The installer may also support older API versions.",
+      "type": "string",
+      "exclusiveMinimum": 0
+    },
+    "additionalTrustBundle": {
+      "description": "a PEM-encoded X.509 certificate bundle that will be added to the nodes' trusted certificate store. This trust bundle may also be used when a proxy has been configured",
+      "type": "string"
+    },
+    "additionalTrustBundlePolicy": {
+      "description": "determines when to add the AdditionalTrustBundle to the nodes'' trusted certificate store. \"Proxyonly\" is the default. The field can be set to following specified values. \"Proxyonly\": adds the AdditionalTrustBundle to nodes when http/https proxy is configured. \"Always\": always adds AdditionalTrustBundle.",
+      "type": "string",
+      "enum": [
+        "Proxyonly",
+        "Always"
+      ],
+      "default": "Proxyonly"
+    },
+    "baseDomain": {
+      "description": "The base domain to which the cluster should belong.",
+      "type": "string",
+      "exclusiveMinimum": 0
+    },
+    "capabilities": {
+      "description": "Capabilities configures the installation of optional core cluster components.",
+      "type": "object",
+      "properties": {
+        "baselineCapabilitySet": {
+          "description": "Selects an initial set of optional capabilities to enable. The default value is vCurrent",
+          "type": "string",
+          "default": "vCurrent"
+        },
+        "additionalEnabledCapabilities": {
+          "description": "",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      },
+      "additionalProperties": false
+    },
+    "controlPlane": {
+      "description": "The configuration for the machines that comprise the control plane.",
+      "$ref": "#/definitions/MachinePool"
+    },
+    "compute": {
+      "description": "The configuration for the machines that comprise the compute nodes.",
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/MachinePool"
+      }
+    },
+    "featureGates": {
+      "type": "array",
+      "items": {
+        "type": "string"
+      }
+    },
+    "featureSet": {
+      "description": "A feature set as defined in the OpenShift API. Notably, TechPreviewNoUpgrade can be used to enable Installer features protected by a Tech Preview feature gate. Feature sets will be applied to the cluster--not just the Installer--and may affect the supportability and upgradability of the cluster, depending on the specified feature set.",
+      "type": "string"
+    },
+    "fips": {
+      "description": "Enables FIPS mode (default false)",
```

Comment:
Do we need to encode the default in the description? Isn't it obvious from the `default` attribute?